moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -829,27 +841,22 @@ const LinkModel* RobotState::getRigidlyConnectedParentLinkModel(const std::strin
       *transform = body->getPose() * *transform;
     link = body->getAttachedLink();
   }
-  else if (hasAttachedBody(frame))
-  {
-    auto body{ getAttachedBody(frame) };
-    if (transform)
-      *transform = body->getPose();
-    link = body->getAttachedLink();
-  }
-  else if (getRobotModel()->hasLinkModel(frame))
-  {
-    link = getLinkModel(frame);
-    if (transform)
-      transform->setIdentity();
-    if (!link)
-      return nullptr;
-  }
+  else
+    return nullptr;
+
   // link is valid and transform describes pose of frame w.r.t. global frame
   Eigen::Isometry3d link_transform;
-  auto* parent = getRobotModel()->getRigidlyConnectedParentLinkModel(link, link_transform, jmg);
-  if (parent && transform)
+  const moveit::core::LinkModel* parent;
+  if (transform)
+  {
+    parent = getRobotModel()->getRigidlyConnectedParentLinkModel(link, link_transform, jmg);
     // prepend link_transform to get transform from parent link to frame
     *transform = link_transform * *transform;
+  }
+  else
+  {
+    parent = getRobotModel()->getRigidlyConnectedParentLinkModel(link, jmg);
+  }
```

Comment:
This change attempts to overoptimize. However, the call w/o the transform argument falls back to the original call anyway:
https://github.com/ros-planning/moveit/blob/286f6b3bb4d89475e5300b4260946e631d212237/moveit_core/robot_model/include/moveit/robot_model/robot_model.h#L276-L281